packages/fluentui/docs/src/prototypes/VirtualizedStickyTree/VirtualizedStickyTree.tsx

```diff
@@ -0,0 +1,472 @@
+      })}
+      {children}
+      {stickyBottomItemsIndex.map((itemIndex: number, index: number, arr: number[]) => {
+        const item = renderedItems[itemIndex];
+        return React.cloneElement(item, {
+          style: {
+            position: 'sticky',
+            height: stickyItemSize,
+            zIndex: 111,
+            // stick items to the container bottom, with the smaller-indexed items stacked on bigger-indexed items
+            top: height - (arr.length - index) * stickyItemSize,
+          },
+          onTitleClick: onClickSticky,
+          onFocusParent: data.handleFocusParent,
+          onFocusFirstChild: data.handleFocusFirstChild,
+        });
+      })}
+    </div>
+  );
+};
+
+const calcRenderedHeight = (renderedItems: React.ReactElement[], stickyItemSize: number, itemSize: number) => {
+  let renderedHeight = 0;
+  renderedItems.forEach(item => {
+    renderedHeight = renderedHeight + (item.props.level === 1 ? stickyItemSize : itemSize);
+  });
+  return renderedHeight;
+};
+
+interface ReRenderStickyTreeProps {
+  renderedItems: React.ReactElement<TreeItemProps>[];
+  stickyItemSize: number;
+  itemSize: number;
+  height: number;
+  toplvlItemsId: string[];
+  onActiveItemsIdChange: (idsToRemove?: string[], idsToAdd?: string[]) => void;
+}
+
+interface ItemDataType {
+  renderedItems: React.ReactElement<TreeItemProps>[];
+  isNonSticky: (item: React.ReactElement<TreeItemProps>) => boolean;
+  onClick: (event: React.MouseEvent<HTMLElement>, itemProps: TreeItemProps) => void;
+  handleOnFocus: (event: React.FocusEvent<HTMLElement>) => void;
+  handleFocusParent: ComponentEventHandler<TreeItemProps>;
+  handleFocusFirstChild: ComponentEventHandler<TreeItemProps>;
+  handleOnSiblingsExpand: ComponentEventHandler<TreeItemProps>;
+}
+
+const ReRenderStickyTree = ({
+  renderedItems,
+  stickyItemSize,
+  itemSize,
+  height,
+  toplvlItemsId,
+  onActiveItemsIdChange,
+}: ReRenderStickyTreeProps) => {
+  const [stickyTopIds, setStickyTopIds] = React.useState<string[]>([toplvlItemsId[0]]);
+  const [stickyBottomIds, setStickyBottomIds] = React.useState<string[]>(toplvlItemsId.slice(1));
+
+  const passOnStickyTopIds = React.useCallback((newStickyTopIds: string[]) => {
+    setStickyTopIds([...newStickyTopIds]);
+  }, []);
+  const passOnStickyBottomIds = React.useCallback((newStickyBottomIds: string[]) => {
+    setStickyBottomIds([...newStickyBottomIds]);
+  }, []);
+
+  // when navigate with keyboard, use `itemIdTobeFocused` to remember the focused item id,
+  // then when DOM updates are finished, sticky tree knows which item to focus on
+  const [itemIdTobeFocused, setItemIdTobeFocused] = React.useState<string>(null);
+  const passItemIdTobeFocused = React.useCallback((newItemIdTobeFocused: string) => {
+    setItemIdTobeFocused(newItemIdTobeFocused);
+  }, []);
+
+  React.useEffect(() => {
+    // retain focus after sticky items gotten clicked and triggers DOM updates
+    if (itemIdTobeFocused === null) return;
```

Comment:
use either falsy check or `==` you don't handle `undefined here`

```suggestion
    if (!itemIdTobeFocused) return;
```

```suggestion
    if (itemIdTobeFocused ==  null) return;
```